hazelcast/src/main/java/com/hazelcast/map/impl/proxy/NearCachedMapProxyImpl.java

```diff
@@ -545,12 +545,15 @@ public void executeOnEntriesInternal(EntryProcessor entryProcessor, Predicate pr
     }
 
     @Override
-    protected boolean preDestroy() {
-        if (invalidateOnChange) {
-            mapNearCacheManager.deregisterRepairingHandler(name);
-            removeEntryListener(invalidationListenerId);
+    protected void postDestroy() {
+        try {
+            if (invalidateOnChange) {
```

Comment:
Same here. Wouldn't it make more sense to first `preDestroy()` the normal proxy, then the (extended) Near Cache proxy states?